median/report_codec.go

```diff
@@ -0,0 +1,49 @@
+package median
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+const typeName = "MedianReport"
+
+func newReportCodec(codec types.Codec) (median.ReportCodec, error) {
+	if codec == nil {
+		return nil, errors.New("codec cannot be nil")
+	}
+	return &reportCodec{codec: codec}, nil
+}
+
+type reportCodec struct {
+	codec types.Codec
+}
+
+var _ median.ReportCodec = reportCodec{}
+
+func (r reportCodec) BuildReport(observations []median.ParsedAttributedObservation) (ocrtypes.Report, error) {
+	if len(observations) == 0 {
```

Comment:
why not  check this in `aggregate()`

Reply:
The top line of all the build reports today are those.  This way, aggregate doesn't need to return an error.